classes/privacy/provider.php

```diff
@@ -33,23 +35,70 @@
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 class provider implements
-        // This portfolio plugin does not store any data itself.
-        // It has no database tables, and it purely acts as a conduit, sending data externally.
-        \core_privacy\local\metadata\provider {
+        \core_privacy\local\metadata\provider,
+        \core_privacy\local\request\plugin\provider {
     /**
      * Returns meta data about this plugin.
      *
-     * @param   collection $collection The initialised collection to add items to.
-     * @return  collection A listing of user data stored through this plugin.
+     * @param collection $collection The initialized collection to add items to.
+     * @return collection A listing of user data stored through this plugin.
      */
     public static function get_metadata(collection $collection): collection {
-
         $collection->add_external_location_link('annoto', [
             'userid' => 'privacy:metadata:annoto:userid',
             'fullname' => 'privacy:metadata:annoto:fullname',
             'email' => 'privacy:metadata:annoto:email',
-          ], 'privacy:metadata:annoto');
+        ], 'privacy:metadata:annoto');
+
+        $collection->add_database_table(
+            'local_annoto_completiondata',
+            [
+                'userid' => 'privacy:metadata:local_annoto_completiondata:userid',
```

Comment:
According to https://moodledev.io/docs/4.3/apis/subsystems/privacy#personal-data-in-moodle
not only userId need to be declered here.
in addition, if it's declared, export and other methods needs to be implemented for that data.
(regardless of the data stored on external service which is ok:
https://moodledev.io/docs/4.4/apis/subsystems/privacy/faq#my-plugin-only-sends-data-to-an-external-location-but-doesnt-store-it-locally-in-moodle---what-should-i-do